pkg/controller/node/nodecontroller.go

```diff
@@ -236,6 +236,7 @@ func NewNodeController(
 	podInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    nc.maybeDeleteTerminatingPod,
 		UpdateFunc: func(_, obj interface{}) { nc.maybeDeleteTerminatingPod(obj) },
+		DeleteFunc: nc.maybeDeleteTerminatingPod,
```

Comment:
I don't understand this line. If we observed deletion of pod, why we would even want to try delete it again?